sql/core/src/test/scala/org/apache/spark/sql/sources/InsertSuite.scala

```diff
@@ -654,6 +654,30 @@ class InsertSuite extends DataSourceTest with SharedSparkSession {
     }
   }
 
+  test("dynamic partition overwrite with limitation") {
+    withSQLConf(
+      SQLConf.PARTITION_OVERWRITE_MODE.key -> PartitionOverwriteMode.DYNAMIC.toString,
+      SQLConf.MAX_DYNAMIC_PARTITIONS.key -> "3") {
+      withTable("t") {
+        sql(
+          """
+            |create table t(i int, part1 int, part2 int) using parquet
+            |partitioned by (part1, part2)
+          """.stripMargin)
+
+        sql("insert into t partition(part1=1, part2=1) select 1")
+        checkAnswer(spark.table("t"), Row(1, 1, 1))
+
+        val e = intercept[SparkException] {
+          sql("insert overwrite table t partition(part1=2, part2)" +
+            " select 2, explode(array(2, 3, 4, 5))")
```

Comment:
if this sql was split into multiple individual inserts, will the exception be generated. ? I guess not, as per the code.  My thoughts are echoing the [comment](https://github.com/apache/spark/pull/25840/files#r325981618). However, i feel the expectation should have been an exception be raised when more than 3 partitions are created, whenever it happens.

Reply:
> if this sql was split into multiple individual inserts

If you means Insert into [multiple tables](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DML#LanguageManualDML-InsertingdataintoHiveTablesfromqueries) in one sql, the check no needs to aggregate the total partition number. Each Insert reachs the limitation should fail fast.

> i feel the expectation should have been an exception be raised when more than 3 partitions are created, whenever it happens.
I've updated the UT, has it addressed your question?